### PR TITLE
Fix where op

### PR DESCRIPTION
**Pull request by AakashKumarNain**

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! What is the nature of the fix? Can you add a unit test to demonstrate it?

**Comment by AakashKumarNain** [comment]
@fchollet the two inputs `x1` and `x2` should have been marked as optional from the start because `where` works with condition alone in which it acts similar to `np.nonzero()`. Earlier `ops.where(cond, None, None)` worked fine for TF and JAX but in torch backend, it was converting the two arguments to tensors, and hence failing. This is a fix for making these arguments optional.

Though I don't think it requires any additional tests, but if you want I can add some

**Comment by fchollet** [comment]
> Though I don't think it requires any additional tests, but if you want I can add some

Please do, that behavior should be tested.

**Review by fchollet — COMMENTED** [review]
Thanks for the update! The numpy test seems to be failing, please take a look. https://github.com/keras-team/keras-core/actions/runs/5941786394/job/16113390567?pr=770

**Comment by AakashKumarNain** [comment]
@fchollet I have simplified the code a bit more, and corrected the behavior with numpy backend. Please let me know if you want any more changes

File: keras_core/ops/numpy.py [hunk 1]
```diff
@@ -5033,7 +5033,12 @@ def vstack(xs):
 
 
 class Where(Operation):
-    def call(self, condition, x1, x2):
+    def call(self, condition, x1=None, x2=None):
+        if (x1 is None and x2 is not None) or (x1 is not None and x2 is None):
+            raise ValueError(
+                "`x1` and `x2` either both should be `None`"
+                + " or both should have non-None value."
```
Comment (fchollet) [inline keras_core/ops/numpy.py:5040]: You don't need this error message 2x, just keep the one in `where`

File: keras_core/ops/numpy.py [hunk 2]
```diff
@@ -5058,6 +5064,11 @@ def where(condition, x1, x2):
         A tensor with elements from `x1` where `condition` is `True`, and
         elements from `x2` where `condition` is `False`.
     """
+    if (x1 is None and x2 is not None) or (x1 is not None and x2 is None):
+        raise ValueError(
+            "`x1` and `x2` either both should be `None`"
+            + " or both should have non-None value."
```
Comment (fchollet) [inline keras_core/ops/numpy.py:5070]: You can use implicitly string concatenation, not `+`

**Pull request merged**